### DISCUSSION TITLE
MSBuildWorkspace analysis ignores custom .editorconfig values?

### DISCUSSION THREAD

**Discussion by franss22** [dotnet/roslyn#70870] [Q&A]
I'm doing a command-line tool that runs a specific set of analyzers over a specified solution.

I create a MSBuildWorkspace with the solution file, then get the solution. 

```cs
MSBuildLocator.RegisterDefaults();
using (var workspace = MSBuildWorkspace.Create())
{
    var solution = await TryOpenSolutionAsync(workspace, options.Solution);
    var compilations = await Task.WhenAll(solution.Projects.Select(p => p.GetCompilationAsync()));
    foreach (var compilation in compilations)
    {
        var analyzerTask = compilation.WithAnalyzers(analyzers).GetAnalyzerDiagnosticsAsync();
        //awaits and saves the analyzer results in a CSV
    }
}
```

However, while it does respect the severity settings defined in the solution's .editorconfig files, some of my analyzers use some custom key/value pairs stored in .editorconfig. Specifically, one of the analyzers looks for a whitelisted string literal.
```
dotnet_diagnostic.MysteryGuest.IgnoredFiles = whitelisteFilePath
```
This works fine when I load my analyzers on an editor, but when I run the analysis from the command line tool, it does not respect this rule. While debugging i found the following:
- When the analyzer looks for the value in the `AnalyzerConfigOptions` object with `TryGetValue`, it gets a null value.
- When checking the project field AnalyzerConfigDocuments field, it does not have them in `AdditionalDocuments` (@CyrusNajmabadi  told me it should be there to be accessible from an analyzer)

![image](https://github.com/dotnet/roslyn/assets/52766048/c7800487-8636-4e7c-94d0-7fa2d34ad4ca)

Am I doing it wrong, or is it a problem with MSBuildWorkspace? And if I'm doing it wrong, what should I be doing?

**Comment by CyrusNajmabadi** [comment]
@jasonmalinowski (or pull in anyone you think knows MSBuildWorkspace well).  Who is responsible for adding .editorconfig to the workspace?  Should the MSBuildWorkspace do this?

**Reply by jasonmalinowski** [reply]
Yes, that should populate just fine. @franss22 if you look at the MSBuildWorkspace's Diagnostics property are there any errors listed?

**Reply by franss22** [reply]
There are 0 diagnostics in the workspace's property before and after running the analysis
![image](https://github.com/dotnet/roslyn/assets/52766048/1fdd4034-2d78-40e7-a824-1469560a5687)

**Reply by franss22** [reply]
Any idea of what's happening? If it's needed i can show the code with more detail.
